docs/docs/guides/ecosystem/huggingface.md

```diff
@@ -0,0 +1,116 @@
+---
+sidebar_position: 1
+hide_table_of_contents: true
+---
+
+
+# Huggingface
+
+Recording traces of LLM applications in a centralized database is crucial during both development and production phases. These traces are invaluable for debugging and for creating a dataset of challenging examples to test and enhance your application.
+
+You can use models from the Hugging Face ecosystem with Weave to keep track of all your interactions and generate insights.
+
+## Using a models locally
+
+You have fine-tuned a model from the Hugging Face HUB a now you want to interact with it. You can start tracing your model by wrapping the model call with a `weave.op()`:
+
+1. Using the `pipeline` API:
+    ```python
+    import weave
+    from transformers import pipeline
+
+    @weave.op()
+    def generate(model_name: str, prompt: str, temperature: float = 0.7) -> str:
+        messages = [
+            {"role": "user", "content": prompt},
+        ]
+        pipe = pipeline("text-generation", model=model_name, temperature=temperature)
+        return pipe(messages)
+    ```
+    
+    this is very inefficient as the model is instantiated every time the function is called.
+
+2. Explicitly using the `AutoModel` and `generate` methods:
+    ```python
+    import weave
+    from transformers import AutoModel, AutoTokenizer
+
+    model = AutoModel.from_pretrained(model_name)
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+    @weave.op
+    def generate(prompt: str, temperature: float = 0.7) -> str:
+        messages = [{"role": "user", "content": prompt}]
+        input_ids = tokenizer.apply_chat_template(messages, return_tensors="pt").to(model.device)
+        output = model.generate(input_ids, temperature=temperature, do_sample=True)
+        return outputs[0][input_ids.shape[-1]:] # we remove the input from the generation
+    ```
+
+3. Explicitly using the `AutoModel` and `generate` methods:
```

Comment:
@tcapelle we would need to find a slightly different name for this approach than for Nr. 2 (for now it's the same) - maybe also just skip the Nr. 2 and only show this one to push the best practice of having an explicit prompt formatting step?

Reply:
push push, feel free to make any change